app/react/settings/users/UserForm.jsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import PropTypes from "prop-types";
+import InlineLoading from "react/common/ui/InlineLoading";
+
+class UserForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            email: "",
+            password: "",
+            realname: "",
+            role: "Select a role"
+        };
+
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.renderRoles = this.renderRoles.bind(this);
+    }
+    handleSubmit(e) {
+        e.preventDefault();
+        const user = Object.assign({}, this.state);
+        this.props.saveNewUser(user);
+        // after success, clear state
+        // after failure, state remains
+        // TBD after we figure out endpoints
+    }
+
+    handleChange(event) {
```

Comment:
I think this would be cleaner as a method per field ie `handlerEmailChange` etc. It'll remove the entire switch statement. Or you could write a generic method to be called inline like
`onChange={event => { this.handleChange('email', event); }}`

Reply:
Will refactor